docs/dev/design/sphinx-jquery.rst

```diff
@@ -0,0 +1,125 @@
+    * https://github.com/pydata/pydata-sphinx-theme/issues/764
+
+This is a "request for comments" for a community-owned Sphinx extension that bundles jQuery.
+
+
+Overview
+--------
+
+Comment deadline:
+  November 1st, 2022
+
+Package-name:
+  ``sphinxcontrib-jquery``
+
+Python package:
+  ``sphinxcontrib.jquery``
+
+Dependencies:
+  Python 3+, Sphinx 1.8+ (or perhaps no lower bound?)
+
+Ownership:
+  Read the Docs core team will implement the initial releases of an otherwise community-owned package that lives in https://github.com/sphinx-contrib/jquery
+
+Functionality:
+  sphinxcontrib-jquery is a Sphinx extension that provides a simple mechanism for other Sphinx extensions and themes to ensure that jQuery is included into the HTML build outputs and loaded in the HTML DOM itself.
+  More specifically, the extension ensures that jQuery is loaded exactly once no matter how many themes and extensions that request to include jQuery nor the version of Sphinx.
+
+Scope:
+  This extension assumes that it's enough to provide a single version of jQuery for all of its dependent extensions and themes.
+  As the name implies, this extension is built to handle jQuery only.
+  It's not a general asset manager and it's not looking to do dependency resolution of jQuery versions.
+
+Usage
+-----
+
+The primary users of this package are
+**a) theme and extension developers** and
+**b) documentation project owners**.
+
+
+Theme and extension developers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following 2 steps need to be completed:
+
+#. A Sphinx theme or extension should depend on the python package ``sphinxcontrib-jquery``.
+#. In your extension's or theme's ``setup(app)``, call ``app.setup_extension("sphinxcontrib.jquery")``.
+
+In addition to this, we recommend extension and theme developers to log to the browser's ``console.error`` in case jQuery isn't found. The log message could for instance say::
+
+  if ($ == "undefined") console.error("<package-name> depends on sphinxcontrib-jquery. Please ensure that <package-name>.setup(app) is called or add 'sphinxcontrib-jquery' to your conf.py extensions setting.")
+
+
+Documentation project owners
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you are depending on a theme or extension that did not itself address the removal of jQuery from Sphinx 6, you can patch up your project like this:
+
+#. Add ``sphinxcontrib-jquery`` to your ``requirements.txt``.
+#. Add ``sphinxcontrib.jquery`` to your ``extensions`` setting in ``conf.py``.
+
+
+Calling ``app.setup_extension("sphinxcontrib.jquery")``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When a Sphinx theme or extension calls `setup_extension() <https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.setup_extension>`_, a call to ``sphinxcontrib.jquery.setup(app)`` will happen. Adding ``sphinxcontrib.jquery`` to a documentation project's ``conf.extensions`` will also call ``sphinxcontrib.jquery.setup(app)`` (at most once).
```

Comment:
This paragraph is confuses me a little. It seems that even doing both, calling `setup_extension()` _and_ adding it to `conf.extensions` will call the function _just once_ which I understand it's not how it works.

I would rephrase it as:

```suggestion
Either calling `setup_extension() <https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.setup_extension>`_ from the theme/extension or adding ``sphinxcontrib.jquery`` to a documentation project's ``conf.extensions``  will produce a call to ``sphinxcontrib.jquery.setup(app)`` (at most once).
```

Reply:
These are different patterns.

One pattern is for extension owners (that call `setup_extension`), the other pattern is for project owners that write a `conf.py`.

> which I understand it's not how it works.

That's how it works though :) It's even implemented here:

https://github.com/sphinx-contrib/jquery/blob/master/sphinxcontrib/jquery.py#L8

Reply:
Awesome! Thanks 👍🏼